numpy/_core/include/numpy/dtype_api.h

```diff
@@ -477,4 +481,18 @@ typedef PyArray_Descr *(PyArrayDTypeMeta_FinalizeDescriptor)(PyArray_Descr *dtyp
 typedef int(PyArrayDTypeMeta_SetItem)(PyArray_Descr *, PyObject *, char *);
 typedef PyObject *(PyArrayDTypeMeta_GetItem)(PyArray_Descr *, char *);
 
+typedef int (PyArray_CompareFuncWithDescr)(const void *, const void *,
+                                           PyArray_Descr *);
```

Comment:
The naming is a bit weird here, but I didn't want to disturb the original type as it's used a lot. I think the `SortCompareFunc` should still be a unique type so will do that (even if only a clone of this type).

Reply:
IMO this is fine, if only because it exists right now 😄